### PR TITLE
Podcast Promo Prototype - Positioning Rules Update

### LINKED ISSUE #9463: (Prototype)Add podcast promo at or after the 8th paragraph when there is enough text
Is your feature request related to a problem? Please describe.
Wrap text around the podcast promo
The promo will be inserted in to the article depending on the following rules
* The promo will only appear from the 8th paragraph. This means the promo can appear next to the 8th paragraph or any text paragraph after.
* The promo will only appear when the continuous char count for a paragraph (or multiple paragraphs) is greater than 780 chars (no spaces) or 940 (with spaces). The text count is for paragraph type blocks only 
* For all other situations, don't show the promo 

* This is based on a brasil article with normal zoom and screen with of 1007px (psammead grid group 3 max width).
https://www.test.bbc.com/portuguese/internacional-58718659?renderer_env=live
image

UX: https://zpl.io/VxAR96E

**Describe alternatives you've considered**

**Testing notes**
Podcast promo showing at 8th paragraph
Given that I'm viewing a screen with a width of equal or greater than 400 and less than 1008 (psammead group 2+)
And the article type is a CPS Story
When I view the article
Then the Podcast promo in the body of the article
And the podcast promo is next to the 8th paragraph
And the Text wraps around the promo image
Example page:  http://www.test.bbc.co.uk/mundo/23457515
Example page (multiple paragraphs):  https://www.test.bbc.com/mundo/23457516

Podcast promo showing at 10th paragraph
Given that I'm viewing a screen with a width of equal or greater than 400 and less than 1008 (psammead group 2+)
And the article type is a CPS Story
And there is a continuous text block after paragraph 10
When I view the article
Then the Podcast promo in the body of the article
And the promo is next to the 10th paragraph
And the Text wraps around the promo image
Example page (multiple paragraphs):   https://www.test.bbc.com/mundo/23457517

Podcast promo not showing
Given that I'm viewing a screen with a width of equal or greater than 400 and less than 1008 (psammead group 2+)
And the article type is a CPS Story
And there is no continuous text block 
When I view the article
Then the Podcast in not in the body of the article
Example page: https://www.test.bbc.com/mundo/23457518

[Tester to complete]

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

- [ ] This feature is expected to need manual testing.

**Checklist**
- [ ] (BBC contributors only) This issue follows the [repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)

**Additional context**

### PR DESCRIPTION
**Pull request by ryanmccombe**
Resolves https://github.com/bbc/simorgh/issues/9463

**Overall change:**
Update prototype positioning rules - block should now be positioned next to the earliest continuous sequence of text that happens at or after paragraph 8

---

- [ ] I have assigned myself to this PR and the corresponding issues
- [ ] I have added the `cross-team` label to this PR if it requires visibility across World Service teams
- [ ] I have assigned this PR to the Simorgh project
- [ ] (BBC contributors only) This PR follows the [repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)

**Testing:**

- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false yarn test:e2e:interactive`)
- [ ] This PR requires manual testing

### CODE REVIEW

**Review by amoore108 — APPROVED** [review]

File: src/app/routes/cpsAsset/getInitialData/insertPodcastPromo/index.js [hunk 1]
```diff
@@ -5,37 +5,56 @@ import insert from 'ramda/src/insert';
 import slice from 'ramda/src/slice';
 import findNthIndex from '#lib/utilities/findNthIndex';
 
+// Podcast promos are inserted if the page content has a continuous sequence of paragraph blocks
+// These paragraph blocks must have a minimum character count in total to ensure there is
+// enough text to wrap around the promo component
+const MINIMUM_CHARACTER_COUNT = 940;
```
Comment (HarveyPeachey) [inline src/app/routes/cpsAsset/getInitialData/insertPodcastPromo/index.js:11]: Does this minimum number take into account character sizing for the different fonts and sizing of characters in the services that this is being used on?

Comment (ryanmccombe) [reply src/app/routes/cpsAsset/getInitialData/insertPodcastPromo/index.js:11]: It's a pretty rough metric - it's hard to get this number right considering different font sizes and (especially) different screen sizes.  The 940 number came from @JonBeeb - I believe it's a rough average based on the services we're likely to be running this experiment on

File: src/app/routes/cpsAsset/getInitialData/insertPodcastPromo/index.test.js [hunk 2]
```diff
@@ -4,43 +4,47 @@ import path from 'ramda/src/path';
...
+    ${4}     | ${[...Paragraphs(7), P(1000)]}                   | ${7}
+    ${5}     | ${[...Paragraphs(7), P(500), P(500)]}            | ${7}
+    ${6}     | ${[H, ...Paragraphs(7), P(1000)]}                | ${8}
+    ${7}     | ${[...Paragraphs(7), P(500)]}                    | ${-1}
+    ${8}     | ${[...Paragraphs(7), P(500), H, P(500)]}         | ${-1}
+    ${9}     | ${[...Paragraphs(7), P(500), H, P(500), P(500)]} | ${9}
   `(
     '[scenario $scenario]: places podcast promo at correct index',
-    ({ inputBlocks, expectedIndex }) => {
+    ({ scenario, inputBlocks, expectedIndex }) => {
```
Comment (ryanmccombe) [inline src/app/routes/cpsAsset/getInitialData/insertPodcastPromo/index.test.js]:
```suggestion
    ({ inputBlocks, expectedIndex }) => {
```

**Review by mfonofm — APPROVED** [review]
looks good

**Pull request merged**